api/controllers/grades.go

```diff
@@ -242,9 +284,24 @@ func gradesAggregation(flag string, c *gin.Context) {
 
 		// Get valid course ids
 		if number == "" {
+			// If only the prefix is provided, filter on the prefix
 			courseFind = bson.D{{Key: "subject_prefix", Value: prefix}}
 		} else {
-			courseFind = bson.D{{Key: "subject_prefix", Value: prefix}, {Key: "course_number", Value: number}}
+			// Old code that filter on combination of prefix and course number (in case we need it in the future)
+			// courseFind = bson.D{{Key: "subject_prefix", Value: prefix}, {Key: "course_number", Value: number}}
+
+			// If both prefix and course_number are provided, find the associated internal_course_number to filter on
+			sampleCourseQuery = bson.D{
+				{Key: "subject_prefix", Value: prefix},
+				{Key: "course_number", Value: number},
+			}
+			// parse the queried document into the sample course
+			err = courseCollection.FindOne(ctx, sampleCourseQuery).Decode(sampleCourse)
```

Comment:
This I forgot to add & in ```.Decode(&sampleCourse)```

Reply:
You can commit the fix and it'll update the PR accordingly :)